### ISSUE TITLE
Test failure on go1.17

### ISSUE THREAD

**Issue by maxice8** [coredns/coredns#4856]
**What happened**: Tests fail when building with go1.17

**What you expected to happen**: Tests pass

**How to reproduce it (as minimally and precisely as possible)**: run testsuite with go1.17

**Anything else we need to know?**:
```
--- FAIL: TestLookupStaticHost (0.00s)
    hostsfile_test.go:130: 0, lookupStaticHostV4(localhost.) = [127.0.0.1]; want [127.0.0.1 127.0.0.2 127.0.0.3]
--- FAIL: TestLookupStaticAddr (0.00s)
    hostsfile_test.go:217: , lookupStaticAddr(127.0.0.2) = []; want &{{{0 0} 0 0 0 0} [.] 0x3911d50 0x3911d40  {0 0 <nil>} 0 0x3b481d0}
    hostsfile_test.go:217: , lookupStaticAddr(127.0.0.3) = []; want &{{{0 0} 0 0 0 0} [.] 0x3911d50 0x3911d40  {0 0 <nil>} 0 0x3b481d0}
--- FAIL: TestHostCacheModification (0.00s)
    hostsfile_test.go:130: 0, lookupStaticHostV4(localhost.) = [127.0.0.1]; want [127.0.0.1 127.0.0.2 127.0.0.3]
FAIL
FAIL	github.com/coredns/coredns/plugin/hosts	0.025s
```

**Environment**:

- the version of CoreDNS: 1.8.4
- Corefile: N/A
- logs, if applicable: See above
- OS (e.g: `cat /etc/os-release`): Alpine Linux Edge (3.15 Alpha)
- Others:

**Comment by chrisohaver** [comment]
this has been fixed with #4844

**Comment by maxice8** [comment]
Thanks for the quick response and sorry I couldn't find the issue/pull request beforehand
